service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -190,104 +125,51 @@ internal class SøknadsbehandlingServiceIverksettTest {
-            søknadsbehandlingRepo = søknadsbehandlingRepoMock,
-            utbetalingService = utbetalingServiceMock,
-            behandlingMetrics = behandlingMetricsMock,
-            vedtakRepo = vedtakRepoMock,
-            kontrollsamtaleService = kontrollsamtaleServiceMock,
-            observer = statistikkObserver,
+            søknadsbehandlingRepo = mock {
+                on { hent(any()) } doReturn behandling
+            },
+            utbetalingService = mock {
+                on { genererUtbetalingsRequest(any(), any(), any(), any(), any()) } doReturn simulertUtbetaling.right()
+                on { publiserUtbetaling(any()) } doReturn utbetalingsRequest.right()
+            },
+            kontrollsamtaleService = mock {
+                on { opprettPlanlagtKontrollsamtale(any(), any()) } doReturn kontrollsamtale().right()
+            },
         )
+
         val response = serviceAndMocks.søknadsbehandlingService.iverksett(
             SøknadsbehandlingService.IverksettRequest(
                 behandlingId = behandling.id,
-                attestering = Attestering.Iverksatt(attestant, attesteringstidspunkt),
+                attestering = Attestering.Iverksatt(attestant, fixedTidspunkt),
             ),
         )
 
-        val expected = Søknadsbehandling.Iverksatt.Innvilget(
-            id = behandling.id,
-            opprettet = behandling.opprettet,
-            sakId = behandling.sakId,
-            saksnummer = behandling.saksnummer,
-            søknad = behandling.søknad,
-            oppgaveId = behandling.oppgaveId,
-            behandlingsinformasjon = behandling.behandlingsinformasjon,
-            fnr = behandling.fnr,
-            beregning = behandling.beregning,
-            simulering = behandling.simulering,
-            saksbehandler = behandling.saksbehandler,
-            attesteringer = Attesteringshistorikk.empty()
-                .leggTilNyAttestering(Attestering.Iverksatt(attestant, attesteringstidspunkt)),
-            fritekstTilBrev = "",
-            stønadsperiode = behandling.stønadsperiode,
-            grunnlagsdata = Grunnlagsdata.IkkeVurdert,
-            vilkårsvurderinger = Vilkårsvurderinger.Søknadsbehandling.IkkeVurdert,
-        )
+        val expected = søknadsbehandlingIverksattInnvilget().second.copy(id = behandling.id)
 
         response shouldBe expected.right()
 
-        verify(søknadsbehandlingRepoMock).hent(behandling.id)
-        verify(utbetalingServiceMock).genererUtbetalingsRequest(
-            sakId = argThat { it shouldBe sakId },
-            attestant = argThat { it shouldBe attestant },
-            beregning = argThat { it shouldBe TestBeregning },
-            simulering = argThat { it shouldBe simulering },
-            uføregrunnlag = argThat { it shouldBe emptyList() },
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(behandling.id)
+        verify(serviceAndMocks.utbetalingService).genererUtbetalingsRequest(any(), any(), any(), any(), any())
```

Comment:
Prøver å bestemme meg for om vi ikke bryr oss om hva vi sender til utbetaling eller ikke :thinking:

Reply:
Tja, sånn implementasjonen er vil man bare expecte de samme argumentene som man har mocket inn rett over. Argument-matching - kunne man vurdert om man like godt skulle satt det opp i selve mocken? At man bare mocker en funksjon som får id = .. osv

Reply:
Implementasjonen henter 4 felter fra mocken ja og attestanten. Jeg synes det gir mening og teste det. Men det vil jo da feile hvis en refaktorerer det eller endrer parametrene til funksjonen 🤷  Men det er vel akkurat det dere vil unngå?

Reply:
Misforstå meg rett, at noen tester feiler under refaktorering må man forvente. Særlig om man endrer signatur på funksjoner osv. Men at _alle_ tester feiler fordi _alle_ testene tester de samme tingene under panseret er ønskelig å unngå.

```
override fun iverksett(request: SøknadsbehandlingService.IverksettRequest): Either<KunneIkkeIverksette, Søknadsbehandling.Iverksatt> {
        val søknadsbehandling = søknadsbehandlingRepo.hent(request.behandlingId)
            ?: return KunneIkkeIverksette.FantIkkeBehandling.left()

        return forsøkStatusovergang(
            søknadsbehandling = søknadsbehandling,
            statusovergang = Statusovergang.TilIverksatt(request.attestering),
        ).map { iverksattBehandling ->
            when (iverksattBehandling) {
                is Søknadsbehandling.Iverksatt.Innvilget -> {

                    val utbetaling = utbetalingService.genererUtbetalingsRequest(
                        sakId = iverksattBehandling.sakId,
                        attestant = request.attestering.attestant,
                        beregning = iverksattBehandling.beregning,
                        simulering = iverksattBehandling.simulering,
                        uføregrunnlag = iverksattBehandling.vilkårsvurderinger.uføre.grunnlag,
                    )
```
Dette er starten på metoden vi tester. Ved å sjekke argumentene, så tester vi egentlig bare at `forsøkStatusovergang` gjør jobben sin. Og det burde strengt talt være `forsøkStatusovergang`-sitt ansvar, litt på samme måte som vi mocker andre servicer 😄  Jeg kan godt legge tilbake argumentene - det er jo faktisk en trygghet at i fremtiden, så kan metoden endre seg og testen vil da kanskje kunne fange det opp. Men det legger også opp til at testen tester èn ting mer

Reply:
Siden forsøkStatusovergang ikke er mocket, tester vi implisitt den også. Som er like greit synes jeg. Uavhengig om den er mocket eller ikke, så synes jeg fremdeles det gir mening å verifisere hva vi sender inn i `genererUtbetalingsRequest `. Kanskje vi byttet fra å bruke request.attestering til søknadsbehandling sin attestering? Det vil vi ikke avdekke nå. Eller at vi begynte å mutere noen av objektene før vi sender de? Jeg har sett flere service-implementasjoner som bruker copy i kodebasen vår f.eks. Dette vil vi heller ikke avdekke. Men om det er en egen test som sjekker det, spiller ikke så stor rolle for min del.